Allow cluster create and import locations to be overridden

nwmac opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #14012

### Occurred changes and/or fixed issues

This PR adds configuration for the provisioning cluster resource to allow the `create` and `import` locations to be overridden by an extension.

This is then wired into the cluster list and home page views.

Extensions can change the locations - an example use case is where you might want a user to only be able to create or import a single cluster type - for example, adding this in a product file in an extension would only allow them to create/import EKS clusters:

```

export function init(plugin: any, store: any) {
  const {configureType } = plugin.DSL(store, 'manager');

  configureType('provisioning.cattle.io.cluster', {
    custom: {
      createLocation: (params: any) => {
        return {
          name:  'c-cluster-product-resource-create',
          query: {
            type: 'amazoneks'
          },
          params
        }
      },
      importLocation: (params: any) => {
        return {
          name:  'c-cluster-product-resource-create',
          query: {
            mode: 'import',
            type: 'amazoneks'
          },
          params
        }
      },
    }
  });
}
```

This is considered an advanced use case and experimental and will be documented once stable. 

### Areas or cases that should be tested

Testing should focus on testing that the existing UI is not broken by this change.

Both on the home page and the cluster list page, check that clicking the 'Create' and 'Import' buttons takes the user to the default pages allowing them to chose the cluster type.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [x] The PR template has been filled out
- [x] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

aalves08 commented on shell/store/type-map.js:
```diff
@@ -524,6 +525,7 @@ export const getters = {
       depaginate:             false,
       customRoute:            undefined,
       resourceEditMasthead:   true,
+      custom:                 {},
```
This needs to be added to `ConfigureTypeOptions`

aalves08 reviewed (changes requested):
Overall, works as advertised. Was able to override the creation of a cluster for Digital Ocean.

We should add `custom` to the `ConfigureTypeOptions` for Typescript support.

We should also document this feature on the extension docs, stating from which shell version it's available

nwmac commented:
@aalves08 I have updated and added to the type.

As per comment at top, I was proposing not documenting this for now and once we have finalised this API/approach, we can document. There may be changes we want to make, so I consider this experimental for now.

aalves08 reviewed (approved):
LGTM. Sorry, I didn't notice the comment about the documentation. That's fine by me 🙏
